decidim-core/lib/decidim/form_builder.rb

```diff
@@ -250,6 +250,25 @@ def areas_select(name, collection, options = {})
       select(name, selectables, options)
     end
 
+    # Public: Generates a select field for resources.
+    #
+    # name       - The name of the field (usually resource_type)
+    # collection - A collection of resources (Decidim::ActionLog).
+    #              The options are sorted alphabetically.
+    #
+    # Returns a String.
+    def resources_select(name, collection, options = {})
+      resources =
+        collection
+        .select(:resource_type).distinct
+        .pluck(:resource_type)
+        .map { |r| [I18n.t(r.split("::").last.underscore, scope: "decidim.components.component_order_selector.order"), r] }
+        .reject { |r| r[1].match?(/ParticipatoryProcess|Component|Survey|Result|Assembly|Consultation|DummyResource|missing\stranslation/i) }
```

Comment:
isn't it easier to filter by the wanted resource_types instead? Something like:
```
collection
        .select(:resource_type).where(resource_type: [::Decidim::Proposals::CollaborativeDraft, ::Decidim::Comment, ...]).distinct
        .pluck(:resource_type)
```

Reply:
Actually, yes, but I chose to reject because that way if a new resource_type is added and translated, it will automatically be added to the filter. If this is not a desired behavior, I will change it then to select the mentioned resource_types only.

Reply:
yes please @jesusdb , let's use the simpler query which will probably be also more performant